packages/scene-composer/src/components/panels/scene-components/tag-style/IconPicker/IconPicker.tsx

```diff
@@ -0,0 +1,129 @@
+import { Button, FormField, SpaceBetween, TextContent, TextFilter } from '@awsui/components-react';
+import { IconDefinition, library } from '@fortawesome/fontawesome-svg-core';
+import { faInfo, fas } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import React, { useCallback, useEffect, useState } from 'react';
+
+import './IconPickerUtils/IconPicker-aws-overrides.scss';
+import { IIconPicker } from '../interface';
+
+import { IconGridLayout, tmIconPickerContainer, tmIconPickerPopover } from './IconPickerUtils/IconPickerStyles';
+
+library.add(fas);
+
+export const IconPicker = ({ selectedIcon, onSelectIconChange, iconPickerLabel }: IIconPicker): JSX.Element => {
+  const [filteringText, setFilteringText] = useState<string>('');
+  const [newIcon, setNewIcon] = useState<IconDefinition>(selectedIcon ?? faInfo);
+  const [showPicker, setShowPicker] = useState<boolean>(false);
+  const iconsPerPage = 8; // Number of icons per page
+  const defaultRows = 3; // Number of rows to display by default
+  const icons: IconDefinition[] = Object.values(fas); // Array of all Font Awesome Solid icons
+  const totalPages = Math.ceil(icons.length / iconsPerPage); // Calculate the total number of pages
+  const [currentPage] = useState<number>(1);
+  const [numRows, setNumRows] = useState(defaultRows);
+  const generateRandomString = Math.random().toString(16).slice(2);
+  const [randomDomId] = useState<string>(generateRandomString);
```

Comment:
looks like this component will need to be fully rerendered during each render cycle because all ids will be changed. is there any other identifier to use so we only rerender when needed?

Reply:
I don't think so. I did not want to change the ID of the UI elements for every render but I wanted to use a unique one for different instances of icon picker. State variable won't change for every render unless we update it with set state. I think this is best way to get an random ID without updating a state and changing the active state of the picker.

Reply:
oh, `Math.random()` is called during every render so new `generateRandomString` will be generated, but it won't update the state after initialize

Reply:
Yes. we are not setting this anywhere. This actually good wrt to performance as well.